pkg/operator/ceph/controller/spec.go

```diff
@@ -207,14 +213,20 @@ func RookVolumeMounts(dataPaths *config.DataPathMap, confGeneratedInPod bool) []
 }
 
 // DaemonVolumesBase returns the common / static set of volumes.
-func DaemonVolumesBase(dataPaths *config.DataPathMap, keyringResourceName string) []v1.Volume {
+func DaemonVolumesBase(dataPaths *config.DataPathMap, keyringResourceName string, dataDirHostPath string) []v1.Volume {
 	configOverrideVolume, _ := configOverrideConfigMapVolumeAndMount()
 	vols := []v1.Volume{
 		configOverrideVolume,
 	}
 	if keyringResourceName != "" {
 		vols = append(vols, keyring.Volume().Resource(keyringResourceName))
 	}
+	// data is persisted to host
+	if dataDirHostPath != "" {
+		hostPathType := v1.HostPathDirectoryOrCreate
+		src := v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: dataDirHostPath + daemonSocketsSubPath, Type: &hostPathType}}
```

Comment:
`path.Join()` is generally preferred over the string addition in case separators are needed between paths.
```suggestion
		src := v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: path.Join(dataDirHostPath,  daemonSocketsSubPath), Type: &hostPathType}}
```